python/metashape_workflow.py

```diff
@@ -14,69 +14,68 @@
 # ---- If not running interactively, the config file should be supplied as the command-line argument after the python script, e.g.: python metashape_workflow.py config.yml
 
 
-## Load custom modules and config file: slightly different depending whether running interactively or via command line
+# Load custom modules and config file: slightly different depending whether running interactively or via command line
 try:  # running interactively (in linux) or command line (windows)
-    from python import metashape_workflow_functions as meta
-    from python import read_yaml
+    from python.metashape_workflow_functions import MetashapeWorkflow
 except:  # running from command line (in linux) or interactively (windows)
-    import metashape_workflow_functions as meta
-    import read_yaml
+    from metashape_workflow_functions import MetashapeWorkflow
 
 if sys.stdin.isatty():
     config_file = sys.argv[1]
 else:
     config_file = manual_config_file
 
-## Parse the config file
-cfg = read_yaml.read_yaml(config_file)
+meta = MetashapeWorkflow(config_file)
 
 ### Run the Metashape workflow
 
-doc, log, run_id = meta.project_setup(cfg, config_file)
+meta.project_setup()
 
-meta.enable_and_log_gpu(log, cfg)
+meta.enable_and_log_gpu()
 
-if (cfg["photo_path"] != "") and (cfg["addPhotos"]["enabled"]):  # only add photos if there is a photo directory listed
-    meta.add_photos(doc, cfg)
+if (meta.cfg["photo_path"] != "") and (
+    meta.cfg["addPhotos"]["enabled"]
```

Comment:
I'd consider other options to how we check if a step should be run. As written, it seems to introduce unnecessary complexity into the calling script. One option would be to do this check (and other analogous ones for other processing steps) in the method. In this case, you could log/warn if a step is not run, or do nothing as currently done. If you want explicit checks in the calling script for each step, I'd recommend creating convenience methods such as `add_photos_enabled` that packages up these config checks in a way that's hidden from the user.

Reply:
I could go either way on this. What is the reason to shift the complexity from the calling script to the method? I kind of like that if `add_photos` is called, we know that it is going to add photos, as opposed maybe add, maybe not depending on the config.

Reply:
My thought was if we ended up using this class in multiple places, it would duplicate the functionality of checking whether to run each step and require the user to understand the structure of the config. But I see your point that it's good to be explicit.

Reply:
How about a middle ground where we have a `MetashapeWorkflow.run()` method where the checking logic (and calling of individual component methods) happens, rather than a calling script. This puts the workflow logic in the class, but still means `add_photos` always runs when called, regardless of what's in the cfg.

Reply:
Just to clarify does that mean `metashape_workflow.py` will only instantiate an object of the class and call `MetashapeWorkflow.run()`?

Reply:
This sounds really good to me, I expect that many users who want to use the library would want to run all the steps at once. Then for more complex tasks they can handle the logic of selecting which ones to run themselves.